File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/comment/uncomment/direct/Java/Go/s456888268_commented.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

func solve(s string) int {
	cnt := 0
	ans := 0
	for i := 0; i < len(s); i++ {
		switch s[i] {
		case 'S':
			cnt++
		case 'T':
			if cnt > 0 {
				ans++
				cnt--
			}
		}
	}
	return len(s) - 2*ans
}

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	scanner.Scan()
	input := scanner.Text()
	fmt.Println(solve(input))
}
// End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/comment/uncomment/direct/Java/Python/s456888268_commented.py
def solve():
    tmp = input().strip()
    s = list(tmp)
    n = len(s)
    ans = 0
    cnt = 0
    for c in s:
        if c == 'S':
            cnt += 1
        else:
            if cnt == 0:
                continue
            ans += 1
            cnt -= 1
    print(n - 2 * ans)

if __name__ == '__main__':
    solve()
# End of Code
